compiler/qsc_passes/src/entry_point.rs

```diff
@@ -139,6 +145,9 @@ impl<'a> Visitor<'a> for EntryPointFinder<'a> {
             if item.attrs.iter().any(|a| a == &Attr::EntryPoint) {
                 self.callables.push((callable, item.id));
             }
+            if callable.name.name.as_ref().to_lowercase() == "main" {
```

Comment:
Should we validate that the callable is an operation?

Reply:
If so, we should have another test as the tests in the PR currently always are operations.

Reply:
No, a function can be an entry point today. Granted that entry point can't really do anything interesting as it can't generate any dynamic values and as such is reducible to a constant... but it would be a new restriction that didn't exist before.